### ISSUE TITLE
need option to override dependency in root package (useful for local development)

### ISSUE THREAD

**Issue by timotheecour** [dlang/dub#1378]
often times I face situation where my main dub.json has a dependency A that itself depends on B:X.Y.Z
For whatever reason I need a newer version of B (X'.Y'.Z'). Currently the only way is to git clone A and manually update its dub.json, and then change dependency A in main dub.json to a path dependency pointing to the git clone. It would be nice to simply allow overriding B:X'.Y'.Z' in main dub.json and avoid all this extra work. This would only be allowed locally, not to be pushed to dub registery.
eg syntax:

in main dub.json:
```
"B": ":=1.2.3"
```

**Comment by wilzbach** [comment]
Why don't you bump the version in the dub.selections.json? That's what it's therefore after all...

**Comment by timotheecour** [comment]
but `dub.selections.json` will not even get generated in the 1st place if there's a conflict (IIRC)

**Comment by MartinNowak** [comment]
No, the dependency should declare what versions it is compatible with and people should use less strict requirements, e.g. `~>1.0.0`. Also dependecies (libraries) don't pin their exact version (should not check-in dub.selections.json).
Allowing to mess with dependency specs in dependents would add a lot of additional complexity.

> but dub.selections.json will not even get generated in the 1st place if there's a conflict (IIRC)

What sort of conflict do you mean here? If you package requires a newer version of B, just state so in your `dub.sdl` and the dependency resolver will (try to) find a version that suits both your's and `A` depenency spec.
